### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by thetazero**
This was done to reduce the code duplication drastically. 

From the README:

The folder `frame/` contains the barebones sattelite state machine code, and the `main.py`. 
This code remains independent of whether you are targeting pycubedmini, pycubed, or emulating hardware.

The `drivers` folder contains libaries allowing one to interface with the target hardware.
We currently have the example, pycubedmini and pycubedmini emulation. 
We plan to support pycubed in the future.

The `applications` folder contains the programs that utilize the drivers. 
For example: the actual flight software and the example software.

### CODE REVIEW

**Comment by thetazero** [comment]
Linter is (correctly) flagging a lot of things, but I don't think it's worth trying to refactor every file in this pull.

File: state_machine/example.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # Example Explanation
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/example.md:1]: This is just the title, I was thinking it's explaining the example, hence what I put. It explains both the driver & application sides of it so I don't think `applications/example` is appropriate. What do you think we should name it?

File: state_machine/README.md [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+
+The folder `frame/` contains the `main.py` (entry point for our software) and a custom state machine implementation.
+This code is minimal enough that no changes should be required regardless of if you are targeting pycubedmini, pycubed, or emulating hardware.
+
+The `drivers` folder contains libaries allowing one to interface with the target hardware.
+We currently have the example emulator, pycubedmini driver and pycubedmini emulation. 
+We plan to support pycubed in the future.
 
-To build the flight software you run `sh build.sh {plugin}` where `{plugin}` is the path to the plugin code (depends on hardware/purpose).
+The `applications` folder contains the programs that utilize the drivers to achieve the specific mission objective. Such as detumbling, beacon transmisions, logging, power management, etc...
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/README.md:12]: Good catch, I'll try and clarify it.

File: state_machine/README.md [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
-To build the files run `sh build.sh plugs/emulation` and to run it run `sh run.sh`. 
 
-## Building the actual flight software
-build: `sh build.sh plugs/flight`
-run: copy over files to the CIRCUITPY drive (don't automate with rm -rf *, as this has led to corrupting the file system in the past)
+## Building Flight Software
 
-## Software Architecture Overview
+To build the flight software you run `sh build.sh {driver} {application}`.
+The {driver} is the part of the software that interfaces with the hardware (or emulates it).
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/README.md:28]: good idea

File: state_machine/README.md [hunk 4]
```diff
@@ -1,31 +1,46 @@
...
+The {application} is what the software attempts to achieve the mission objective (by utilizing the driver to communicate with the hardware).
+This allows us to easily test and develop flight software localy by emulating the hardware.
+
+## Simple Example
 
-The folder `frame/` contains the barebones sattelite state machine code, and the `main.py`. 
-This code remains unchanged independent of whether you are using pycubedmini, pycubed, or running your own test script.
+Build and Run: `sh build.sh drivers/example applications/example && sh run.sh`.
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/README.md:34]: That's reasonable, while I'm developing however I almost always just want to copy the entire build and run command. So it is slightly more convenient while developing my way. I'm leaning slightly towards your way, what are your thoughts?

[the remark this answers is not in the document]
Comment (jbwillis) [reply state_machine/README.md:34]: Do you copy this often?

[the remark this answers is not in the document]
Comment (jbwillis) [reply state_machine/README.md:34]: Not worth spending a bunch of time on. I'm fine with it either way, so pick one and go with it.

File: state_machine/example.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbwillis) [reply state_machine/example.md:1]: Oh, I see. I'd say it's fine as is.

File: state_machine/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/README.md:34]: pretty often, especially when I'm refactoring or modifying the frame. Perhaps the real answer is better VSCode integration.

**Review by jbwillis — APPROVED** [review]
This is fantastic, definitely a step in the right direction towards minimizing code duplication!

**Pull request merged**